modules/openapi-generator/src/main/resources/go/client.mustache

```diff
@@ -306,17 +306,22 @@ func (c *APIClient) prepareRequest(
 	return localVarRequest, nil
 }
 
-func (c *APIClient) decode(v interface{}, b []byte, contentType string) (err error) {
+func (c *APIClient) decode(v interface{}, reader io.ReadCloser, contentType string) (err error) {
```

Comment:
the body must always close regardless of content type or it will leak file handles, so we likely need a catch all for any other case. For the case of streams, should make it clear in the docs that the developer must close the body later.

Reply:
yes, but this template was hard to do it
I thing add middleware to cache response is best way e.g
if I want origin response:  
`rsp, err := client.MyTag.SomeRequest(context, data_body).Response()`
if I want package level response
`rsp, err := client.MyTag.SomeCall(context, data_body).Process()`

Reply:
> yes, but this template was hard to do it

Can you elaborate on the challenge? We can add an additional mustache tag for your use case.